ClearKeep/Scenes/Home/Models/CallViewModel.swift

```diff
@@ -0,0 +1,454 @@
+	@Published var remoteViewRenderSize: CGSize = CGSize.zero
+	@Published var isVideoRequesting = false
+	@Published var callType: CallType = .audio
+	@Published var remotesVideoViewConfig = [String : CustomVideoViewConfig]()
+	var remotesVideoViewDict = [String : RTCMTLEAGLVideoView]()
+	var backHandler: (() -> Void)? = nil
+	
+	enum RenderScaleMode {
+		case scaleToFit
+		case scaleToFill
+	}
+	
+	public var callBox: CallBox?
+	var callTimer: Timer?
+	lazy var timeCounter = TimeCounter()
+	
+	var timeoutTimer: Timer?
+	var callInterval: Int = 0
+	
+	override init() {
+		super.init()
+	}
+	
+	func updateCallBox(callBox: CallBox) {
+		self.callBox = callBox
+		self.callType = callBox.type
+		
+		if callBox.type == .audio {
+			self.cameraOn = false
+			self.speakerEnable = false
+		} else {
+			self.cameraOn = callBox.isCamera
+			self.speakerEnable = true
+		}
+		
+		updateMicroConfig()
+		updateSpeakerConfig()
+		updateCameraConfig()
+		
+		updateVideoView()
+
+		self.callBox?.stateDidChange = { [weak self] in
+			DispatchQueue.main.async {
+				let boxStatus = self?.callBox?.status ?? .ended
+				if boxStatus == .answered {
+					self?.startCallTimer()
+					if let type = self?.callBox?.type, type == .video {
+						self?.forceEnableSpeaker()
+					}
+				} else if self?.callStatus != .ringing, boxStatus == .ringing {
+					self?.startCallTimout()
+				} else if boxStatus == .ended {
+					self?.stopCallTimer()
+					self?.stopTimeoutTimer()
+				}
+				self?.callStatus = self?.callBox?.status ?? .ended
+				self?.updateVideoView()
+			}
+		}
+		
+		self.callBox?.renderSizeChangeWithSize = { [weak self] (size, uId) in
+			guard let self = self else {
+				return
+			}
+			print("uId", uId)
+			// Just support only call 1:1 for now
+			self.remoteViewRenderSize = size
+		}
+		
+		self.callBox?.membersInCallDidChange = { [weak self] in
+			guard let self = self else {
+				return
+			}
+			self.updateVideoView()
+		}
+	}
+	
+	func getStatusMessage() -> String {
+		switch callStatus {
+		case .calling:
+			return "Calling..."
+		case .ringing:
+			return "Connecting..."
+		case .ended:
+			return "End call..."
+		default:
+			return ""
+		}
+	}
+	
+	func getUserName() -> String {
+		return callBox?.clientName ?? "N/A"
+	}
+	
+	func getAvatar() -> String {
+		return callBox?.avatar ?? "" //""
+	}
+	
+	func updateVideoView() {
+		DispatchQueue.main.async {
+			if self.localVideoView == nil {
+				self.localVideoView = self.callBox?.videoRoom?.publisher?.videoRenderView
+			}
+			
+			if let listener = self.callBox?.videoRoom?.remotes.values.first {
+				self.remoteVideoView = listener.videoRenderView
+				DispatchQueue.main.asyncAfter(deadline: .now() + 4, execute: {
+					self.remoteVideoView = listener.videoRenderView
+				})
+			}
+			
+			if let isGroup = self.callBox?.isCallGroup {
+				self.callGroup = isGroup
+			}
+			
+			self.remotesVideoView.removeAll()
+			
+			if self.callGroup {
+				let groupId =  self.callBox?.roomRtcId ?? 0
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/fe9f474e9dacdb67166a3b18715e917932515a0f/ClearKeep/Scenes/Home/Models/CallViewModel.swift#L142"></span>Operators should be surrounded by a single whitespace when they are being used.<br />
<code>operator_usage_whitespace</code> <code>CallViewModel.swift:142</code></td>
    </tr>
  </tbody>
</table>